src/FSAL/FSAL_CORTXFS/handle.c

```diff
@@ -2450,27 +2474,30 @@ static fsal_status_t kvsfs_open2_by_handle(struct fsal_obj_handle *obj_hdl,
 	fsal_status_t status = fsalstat(ERR_FSAL_NO_ERROR, 0);
 	cfs_cred_t cred;
 	struct kvsfs_file_state *fd;
-	struct stat stat;
+	cfs_ino_t * ino = NULL;
+	struct stat *stat = NULL;
+	struct cfs_fh * fh = NULL;
 
 	perfc_trace_inii(PFT_OPEN2_BY_HANDLE, PEM_FSAL_TO_NFS);
 	cortxfs_cred_from_op_ctx(&cred);
 	fd = &container_of(state, struct kvsfs_state_fd, state)->kvsfs_fd;
 	obj = container_of(obj_hdl, struct kvsfs_fsal_obj_handle, obj_handle);
+	ino = kvsfs_fh_to_ino(obj->handle);
 
 	status = kvsfs_file_state_open(fd, openflags, obj, false);
 	if (FSAL_IS_ERROR(status)) {
 		goto out;
 	}
 
 	if (attrs_out != NULL) {
-		rc = cfs_getattr(obj->cfs_fs, &cred,
-				 kvsfs_fh_to_ino(obj->handle), &stat);
+		rc = cfs_fh_from_ino(obj->cfs_fs, ino, &fh);
```

Comment:
1. check rc is valid or not before accessing stat.
2. Add a todo comment about creation of FH is temporary.

Reply:
Done.